compiler/rustc_mir_transform/src/coverage/spans.rs

```diff
@@ -76,18 +76,26 @@ pub(super) fn extract_refined_covspans(
 /// invocation, which is unhelpful. Keeping only the first such span seems to
 /// give better mappings, so remove the others.
 ///
+/// Similarly, `await` expands to a branch on the discriminant of `Poll`, which
+/// leads to incorrect coverage if the `Future` is immediately ready (#98712).
+///
 /// (The input spans should be sorted in BCB dominator order, so that the
 /// retained "first" span is likely to dominate the others.)
-fn remove_unwanted_macro_spans(covspans: &mut Vec<SpanFromMir>) {
+fn remove_unwanted_expanded_spans(covspans: &mut Vec<SpanFromMir>) {
+    let mut seen_await_spans = FxHashSet::default();
     let mut seen_macro_spans = FxHashSet::default();
```

Comment:
Question: Is a separate set needed here, or would we get the desired results with just one set?

(I'm not sure what the precise interaction will be between bang-macros and `await` desugaring here.)

Reply:
I'm not sure either. We probably don't need a separate set, I just wasn't sure what to name the combined set.

Reply:
Heh yeah, that might be the trickiest part.

Something like `deduplicated_spans` should be fine.